src/decorators.ts

```diff
@@ -100,6 +105,29 @@ export function observe(targets: string|string[]) {
   }
 }
 
+/**
+ * A TypeScript accessor decorator factory that causes the decorated accessor to
+ * be called when a property changes. `targets` is either a single property
+ * name, or a list of property names.
+ *
+ * This function must be invoked to return a decorator.
+ */
+export function computed<T = any>(...targets: (keyof T)[]) {
+  return (proto: any, propName: string, descriptor: PropertyDescriptor): void => {
+    const targetString = targets.join(',');
+    const propNameCased = propName[0].toUpperCase() + propName.slice(1);
+    const fnName = `__compute${propNameCased}`;
+
+    proto.constructor.prototype[fnName] = descriptor.get;
```

Comment:
I think we should really use `Object.defineProperty` here, in case there's a name collision up the prototype chain.